app_dart/lib/src/request_handlers/get_status.dart

```diff
@@ -30,15 +31,36 @@ class GetStatus extends RequestHandler<Body> {
   final DatastoreServiceProvider datastoreProvider;
   final BuildStatusProvider buildStatusProvider;
 
+  static const String ownerKeyParam = 'ownerKey';
+
   @override
   Future<Body> get() async {
+    final String encodedOwnerKey = request.uri.queryParameters[ownerKeyParam];
+    final DatastoreService datastore = datastoreProvider();
+    final KeyHelper keyHelper = config.keyHelper;
+    final int commitNumber = config.commitNumber;
+
+    /// [timestamp] is initially set as the current time, which allows query return
+    /// latest commit list. If [owerKeyParam] is not empty, [timestamp] will be set
+    /// as the timestamp of that [commit], and the query will return lastest commit
+    /// list whose timestamp is smaller than [timestamp].
+    int timestamp = DateTime.now().millisecondsSinceEpoch;
+
+    if (encodedOwnerKey != null) {
+      final Key ownerKey = keyHelper.decode(encodedOwnerKey);
+      final Commit commit =
+          await datastore.db.lookupValue<Commit>(ownerKey, orElse: () => null);
+
+      timestamp = commit?.timestamp;
+    }
```

Comment:
I think this logic is enough to just be moved its own function

Reply:
Done.